### PR TITLE
Fix issue #558: /jira assign should accept @username

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
Issue link: https://github.com/mattermost/mattermost-plugin-jira/issues/558

### CODE REVIEW

File: server/command_test.go [hunk 1]
```diff
@@ -457,3 +457,79 @@ func TestPlugin_ExecuteCommand_Uninstall(t *testing.T) {
...
+				},
+			},
+			expectedMsgPrefix: "the mentioned user is not connected to Jira",
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			isSendEphemeralPostCalled := false
+			currentTestAPI := api
```
Comment (Kshitij-Katiyar) [inline server/command_test.go:512]: why this variable change ?

File: server/user_test.go [hunk 2]
```diff
@@ -62,3 +64,50 @@ func TestRouteUserStart(t *testing.T) {
...
+		},
+		"non connected user won't appear when mentioned": {
+			mentions: &model.UserMentionMap{
+				"non_connected_user": "non_connected_user",
+			},
+			userSearch:     "non_connected_user",
+			expectedResult: nil,
+			expectedError:  "the mentioned user is not connected to Jira",
+		},
+		"Connected users are shown and returned as JiraUsers when mentioned": {
```
Comment (Kshitij-Katiyar) [inline server/user_test.go:94]:
```suggestion
		"Connected users are shown and returned as Jira Users when mentioned": {
```

File: server/command_test.go [hunk 3]
```diff
@@ -457,3 +457,79 @@ func TestPlugin_ExecuteCommand_Uninstall(t *testing.T) {
...
+
+			p.SetAPI(currentTestAPI)
+			p.instanceStore = p.getMockInstanceStoreKV(1)
+			p.userStore = getMockUserStoreKV()
+
+			cmdResponse, appError := p.ExecuteCommand(&plugin.Context{}, tt.commandArgs)
+			require.Nil(t, appError)
+			require.NotNil(t, cmdResponse)
+			assert.True(t, isSendEphemeralPostCalled)
+			assert.True(t, true)
```
Comment (ayusht2810) [inline server/command_test.go:532]: what are we checking here?

File: server/user_test.go (hunk 2 above)
Comment (ayusht2810) [reply server/user_test.go:94]:
```suggestion
		"Connected users are shown and returned as Jira Users, when mentioned": {
```

File: server/command_test.go [hunk 4]
```diff
@@ -457,3 +457,79 @@ func TestPlugin_ExecuteCommand_Uninstall(t *testing.T) {
...
+					"Expected returned message to start with: \n %s\nActual:\n%s", tt.expectedMsgPrefix, actual)
+			}).Once().Return(&model.Post{})
+
+			p.SetAPI(currentTestAPI)
+			p.instanceStore = p.getMockInstanceStoreKV(1)
+			p.userStore = getMockUserStoreKV()
+
+			cmdResponse, appError := p.ExecuteCommand(&plugin.Context{}, tt.commandArgs)
+			require.Nil(t, appError)
+			require.NotNil(t, cmdResponse)
```
[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply server/command_test.go:530]: discussed

File: server/command_test.go [hunk 5]
```diff
@@ -457,3 +457,78 @@ func TestPlugin_ExecuteCommand_Uninstall(t *testing.T) {
...
+			expectedMsgPrefix: "Please specify an issue key and an assignee search string, in the form `/jira assign <issue-key> <assignee>`",
+		},
+		"assign to valid issue but no user": {
+			commandArgs: &model.CommandArgs{
+				Command: "/jira assign INVALID",
+				UserId:  mockUserIDWithNotifications,
+			},
+			expectedMsgPrefix: "Please specify an issue key and an assignee search string, in the form `/jira assign <issue-key> <assignee>`",
+		},
+		"assign valid issue but non existing user": {
```
Comment (Kshitij-Katiyar) [inline server/command_test.go:491]:
```suggestion
		"assign the valid issue to a nonexisting user": {
```

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/user_test.go [hunk 6]
```diff
@@ -78,36 +84,36 @@ func TestGetJiraUserFromMentions(t *testing.T) {
...
 				"non_connected_user": "non_connected_user",
 			},
-			userSearch:     "non_connected_user",
-			expectedResult: nil,
-			expectedError:  "the mentioned user is not connected to Jira",
+			userSearch:    "non_connected_user",
+			expectedError: "the mentioned user is not connected to Jira",
 		},
-		"Connected users are shown and returned as JiraUsers when mentioned": {
+		"Connected users are shown and returned as Jira Users when mentioned": {
```
Comment (ayusht2810) [inline server/user_test.go:98]:
```suggestion
		"Connected users are shown and returned as Jira Users, when mentioned": {
```

**Review by ayusht2810 — APPROVED** [review]

File: server/command_test.go [hunk 7]
```diff
@@ -457,3 +457,78 @@ func TestPlugin_ExecuteCommand_Uninstall(t *testing.T) {
...
+func TestPlugin_ExecuteCommand_Assign(t *testing.T) {
+	p := &Plugin{}
+	tc := TestConfiguration{}
+	p.updateConfig(func(conf *config) {
+		conf.Secret = tc.Secret
+		conf.mattermostSiteURL = mattermostSiteURL
+	})
+
+	api := &plugintest.API{}
+	api.On("LogError", mock.AnythingOfTypeArgument("string"))
```
Comment (manojmalik20) [inline server/command_test.go:470]: This statement is not testing anything. Use more strict parameter validation.

File: server/issue.go [hunk 8]
```diff
@@ -878,12 +878,17 @@ func (p *Plugin) AssignIssue(instance Instance, mattermostUserID types.ID, issue
 	}
 
 	// Get list of assignable users
-	jiraUsers, err := client.SearchUsersAssignableToIssue(issueKey, userSearch, 10)
-	if StatusCode(err) == 401 {
-		return "You do not have the appropriate permissions to perform this action. Please contact your Jira administrator.", nil
-	}
-	if err != nil {
-		return "", err
+	jiraUsers := []jira.User{}
```
[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply server/issue.go:881]: discussed

[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply server/issue.go:881]: discussed

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**